span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/fieldgenerators/HttpFieldsGenerator.java

```diff
@@ -485,16 +483,18 @@ private void populateUrlParts(Request.Builder requestBuilder) {
 
     String urlStr = requestBuilder.getUrl();
     try {
-      URL url = new URL(urlStr);
-      requestBuilder.setScheme(url.getProtocol());
+      // constructing a URI object here instead of a URL object so that it can handle the cases where url field
+      // contains only path name (like "/customer?customer=392")
+      URI uri = new URI(urlStr);
```

Comment:
I would suggest continuing using URL itself as a URL is a subset of URI, and we can accept many negative scenarios. The problem here is checking if it's a relative URL or an absolute URL and populating fields accordingly.
In the case of a relative URL, URI will also not help in populating host/protocol.

So, we can have a simple function of checking absolute vs relative URL something like,
```
public static boolean isAbsoluteURL(String input)
      throws java.net.MalformedURLException {
    final URL baseHTTP = new URL("http://hypertace.org");
    final URL baseFILE = new URL("file://");
    URL frelative = new URL(baseFILE, input);
    URL hrelative = new URL(baseHTTP, input);
    return frelative.equals(hrelative);
  }
```

Based on the above function,

```
if isAbsoluteURL(urlStr) {
requestBuilder.setScheme(url.getProtocol());
requestBuilder.setHost(url.getProtocol());
} else {
requestBuilder.setScheme("http");
}
```

I have looked at the below scenarios, using URI for some malformed relative URI, we are not getting benefits too.
As an example,

# host present incorrectly
```
  URI("hypertrace.org:1234/customer?customer=392")

  schema:hypertrace.org
  authority:null
  path:null
  query:null
```

# host present but not protocol present
```
  URI("/hypertrace.org:1234/customer?customer=392")
  schema:null
  authority:null
  path:/hypertrace.org:1234/customer
  query:customer=392
```